routes/index.js

```diff
@@ -0,0 +1,141 @@
+  .catch(err => res.status(500).json({status: 'error', message: err, data: null}))
+})
+.post((req, res) => {
+  const input = req.body
+  db.Filter.countDocuments(input)
+      .then(count => {
+        if(count) {
+          res.status(406).json({status: 'error', message: 'Already Exist', data: null})
+        } else {
+          db.Filter.create(input)
+          .then(created => res.status(200).json({status: 'success', message: 'Created', data: {id: created._id, name: created.name}}))
+          .catch(err => res.status(500).json({status: 'error', message: err, data: null}))
+        }
+      })
+      .catch(err => res.status(500).json({status: 'error', message: err, data: null}))
+})
+.delete((req, res) => {
+  const input = {_id: req.body.id}
+  db.Filter.countDocuments(input)
+  .then(count => {
+    if(count) {
+      db.Filter.deleteOne(input)
+      .then(() => res.status(200).json({status: 'success', message: 'Deleted', data: null}))
+      .catch(err => res.status(500).json({status: 'error', message: err, data: null}))
+    } else {
+      res.status(406).json({status: 'error', message: 'Not Found', data: null})
+    }
+  })
+  .catch(err => res.status(500).json({status: 'error', message: err, data: null}))
+})
+.put((req, res) => {
+  const id = {_id: req.body.id}
+  const change = {name: req.body.name}
+  db.Filter.updateOne(id, change)
+  .then(() => res.status(200).json({status: 'success', message: 'Updated', data: null}))
+  .catch(err => res.status(500).json({status: 'error', message: err, data: null}))
+})
+
+
+router.route('/markers')
+.get((req, res) => {
+    const { location, distance, categories } = req.body
+    let query = {
+        location: {
+            $near: {
+                $maxDistance: distance,
+                $geometry: {
+                    type: "Point",
+                    coordinates: location
+                }
+            }
+        }
+    }
+    if (categories.length) query.filters = {$in: categories}
+    db.Pin.find(query).populate('filters').populate('details').exec()
+    .then(foundData => res.status(200).json({
+        status: 'success',
+        message: null,
+        data: foundData.map(found => ({
+            id: found._id,
+            name: found.name,
+            address: found.address,
+            location: found.location.coordinates,
+            categories: found.filters.map(filter => ({id: filter._id, name: filter.name}))
+        }))
+    }))
+    .catch(err => res.status(500).json({status: 'error', message: err, data: null}))
+})
+.post((req, res) => {
+    const newPinDetails = {
+        description: req.body.details.description,
+        images: req.body.images,
+        menu: req.body.details.menu,
+        hours: req.body.details.hours,
+        comments: []
+    }
+    db.Detail.create(newPinDetails)
+    .then(createdDetail => {
+        const newPin = {
```

Comment:
hint: some destructuring might spare us some typing and not repeat that much 😉 
```js
const {name, address, location, categories} = req.body
const newPin = {
    name, 
    address, 
    location, 
    categories, 
    location: {coordinates: location}, 
    details: createdDetail._id
}
```

Reply:
Repeated code is there for future validation